src/routes/api/api-router.ts

```diff
@@ -82,16 +82,23 @@ ApiRouter.post(
 		const offset = Number(req.body.offset) || 0;
 		// only resync installations whose "updatedAt" date is older than x seconds
 		const inactiveForSeconds = Number(req.body.inactiveForSeconds) || undefined;
+		// A date to start fetching commit history(main and branch) from.
+		const commitsFromDate = req.body.commitsFromDate;
 
 		if (!statusTypes && !installationIds && !limit && !inactiveForSeconds){
 			res.status(400).send("please provide at least one of the filter parameters!");
 			return;
 		}
 
+		if (commitsFromDate && !Date.parse(commitsFromDate)) {
```

Comment:
Why aren't you keeping the parsed date to pass it on to the next function?  Better typings than just a string :)

Reply:
also, this should check to see that the date is valid (like, not be in the future with `date < new Date()`)